lib/nmatrix/lapack.rb

```diff
@@ -1,8 +1,20 @@
 module NumRuby::Linalg
-  def self.inv(obj)
-    if obj.is_a?(NMatrix)
-      return obj.invert
+  def self.inv(matrix)
+    if not matrix.is_a?(NMatrix)
+      raise("Invalid matrix. Not of type NMatrix.")
+    end
+    if matrix.dim != 2
+      raise("Invalid shape of matrix. Should be 2.")
+    end
+    if matrix.shape[0] != matrix.shape[1]
```

Comment:
what do you want validate here?

Reply:
Square matrix, I think

Reply:
so I think you could put this in a function as:
```rb
def square_matrix?(matrix)
  raise("Invalid shape. Expected square matrix.") unless matrix.shape[0] != matrix.shape[1]
end
```
WDYT? 🤔